clcache.py

```diff
@@ -74,10 +80,15 @@ def filesBeneath(path):
 
 
 def childDirectories(path, absolute=True):
-    for entry in os.listdir(path):
-        absPath = os.path.join(path, entry)
-        if os.path.isdir(absPath):
-            yield absPath if absolute else entry
+    supportsScandir = (LIST != os.listdir)
+    for entry in LIST(path):
+        if supportsScandir:
+            if entry.is_dir():
+                yield entry.path if absolute else entry.name
+        else:
+            absPath = os.path.join(path, entry)
+            if os.path.isdir(absPath):
+                yield absPath if absolute else entry
```

Comment:
`LIST` is more than just a boolean flag. If we want to support `scandir` module (`LIST` could be reference to `scandir.scandir` or `os.scandir`) . If we don't want, I can simplify code a little bit.

So, I can move check outside of `for` loop, I don't think that it's expensive to check boolean variable, but I can do it if you want.

`supportsScandir = (LIST != os.listdir)` could be outside of `childDirectories`, of course. I can do it in such way:
```python
supportsScandir = True
try:
    import scandir # pylint: disable=wrong-import-position
    WALK = scandir.walk
    LIST = scandir.scandir
except ImportError:
    WALK = os.walk
    try:
        LIST = os.scandir # pylint: disable=no-name-in-module
    except AttributeError:
        LIST = os.listdir
        supportsScandir = False
```